consensus/reactor.go

```diff
@@ -177,6 +177,13 @@ func (conR *ConsensusReactor) AddPeer(peer p2p.Peer) {
 	}
 }
 
+func (conR *ConsensusReactor) InitPeer(peer p2p.Peer) p2p.Peer {
+	// Create peerState for peer
+	peerState := NewPeerState(peer).SetLogger(conR.Logger)
+	peer.Set(types.PeerStateKey, peerState)
+	return peer
+}
+
```

Comment:
Don't we need to remove `peer.Set(types.PeerStateKey, peerState)` from `AddPeer`? Otherwise it will override whenever was written between `InitPeer` and `AddPeer` calls.

Reply:
right. Remove it from AddPeer